stdlib/filename.mli

```diff
@@ -159,6 +159,21 @@ val open_temp_file :
    @before 3.11.2 no ?temp_dir optional argument
 *)
 
+val temp_dir : ?temp_dir: string -> string -> string -> string
+(** [temp_dir prefix suffix] returns the name of a
+   fresh temporary directory inside the temporary directory.
```

Comment:
I thought the idea was to rather tell people to use `Unix` than add more system stuff of that kind to the stdlib. Oh well, the `Filename` module is kind of a lost cause anyways.

In any case:

1. I think it would be good to add an optional argument `perms` to specifiy the permissions.
2. It would be good to significantly rework this doc string which is very confusing, 
   in particular you talk about two temporary directories regardless. I would start with: 
   ```
    [temp_dir ~perms ~temp_dir prefix suffix] creates and returns a fresh directory with 
    permissions [perms] (defaults to 0o700) inside [temp_dir]. …`.
   ```
3. Mention that the function fails if [temp_dir] does not exist (i.e. this does not create the path).

Reply:
> It would be good to significantly rework this doc string which is very confusing,

I agree, but then both the docstrings of `temp_file` and `temp_dir` should be reworked in parallel, as they are facsimiles of each other.

Reply:
Well let's first have a good one for this one.